cogs/onmyoji.py

```diff
@@ -285,6 +323,376 @@ def shiki_bounty_embed(self, shiki):
+
+    @shard.command(name="have")
+    async def shard_set_have(self,ctx,*,args=None):
+        """
+        Returns the list of shards you have.
+        If shards are listed, assigns those shards to your have list.
+        """
+        if not args:
+            try: 
+                have_list = '\n'.join(self.shard_print_have_list(str(ctx.message.author.id)))
+                await ctx.send(f"Shards you **have**: ```\n{have_list}```")
+                return
+            except KeyError:
+                await ctx.send('You do not have a "Have" list yet! Use `&shard` to generate your entry first!')
+                return
+        arg_list = args.split("\n")
+        for shiki in arg_list:
+            numbers, shiki = self.shard_split_variable(shiki)
+            if "frog" not in shiki.lower().strip():
+                if shiki.lower().strip() not in self.shikigami_class.keys():
+                    return await ctx.send(f"The following shikigami is not present in the master Shikigami database: \n**{shiki}**\nSpelling is important, else searches won't work. Please try again.")
+        self.shard_load_json()
+        try: 
+            self.shard_trading_db[str(ctx.message.author.id)]['have'] = arg_list
+        except KeyError:
+            self.shard_trading_db[str(ctx.message.author.id)] = {}
+            self.shard_trading_db[str(ctx.message.author.id)]['have'] = arg_list
+        self.shard_file_writeout()
+        arg_string = '\n'.join(self.shard_trading_db[str(ctx.message.author.id)]['have'])
+        if self.check_trading_status(ctx.author.id) == True:
+            trading_status = "available"
+        else:
+            trading_status = "unavailable"
+        await ctx.send(f'The shards you have are now set to: ```\n{arg_string}```\nYou are currently {bold(trading_status)} for trading.')
+
+    @shard.command(name="notes")
+    async def shard_set_notes(self, ctx, *notes):
+        """
+        Sets the user note for their shard trading database entry.
+        Leave the field blank to disable, otherwise type in a note to set that note.
+        """
+        args = ' '.join(notes)
+        self.shard_load_json()
+        if self.check_trading_status(ctx.author.id) == True:
+            trading_status = "available"
+        else:
+            trading_status = "unavailable"
+        if not args:
+            self.shard_trading_db[str(ctx.author.id)]['notes'] = ''
+            await ctx.send(f"Notes disabled.\nYou are currently {bold(trading_status)} for trading.")
+            self.shard_file_writeout()
+        else:
+            self.shard_trading_db[str(ctx.author.id)]['notes'] = args
+            await ctx.send(f"Your Shard Trading entry Notes have been set to:\n```{args}```\nYou are currently {bold(trading_status)} for trading.")
+            self.shard_file_writeout()
+
+    def check_trading_status(self, user):
+        try:
+            return self.shard_trading_db[str(user)]['status']
```

Comment:
```suggestion
            return 'status' in self.shard_trading_db[str(user)]
```

or if you were checking that the user is in there
```suggestion
            return str(user) in self.shard_trading_db and 'status' in self.shard_trading_db[str(user)]
```
Don't use exceptions to check for things, i forget why but i know its bad so just trust me